modules/gatk4/addorreplacereadgroups/main.nf

```diff
@@ -0,0 +1,47 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+params.options = [:]
+options        = initOptions(params.options)
+
+process GATK4_ADDORREPLACEREADGROUPS {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+    conda (params.enable_conda ? "bioconda::gatk4=4.2.3.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/gatk4:4.2.3.0--hdfd78af_0"
+    } else {
+        container "quay.io/biocontainers/gatk4:4.2.3.0--hdfd78af_0"
+    }
+
+    input:
+    tuple val(meta), path(bam)
+
+    output:
+    tuple val(meta), path("*.bam"), emit: bam
+    path "version.yml",             emit: versions
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+
+    """
+    gatk AddOrReplaceReadGroups" \\
+        -I $bam \\
+        -O ${prefix}.bam \\
+        -ID=4 \\
+        -LB=lib1 \\
+        -PL=ILLUMINA \\
+        -PU=Unit1 \\
+        -SM=20 \\
```

Comment:
These should be provided as `input` options to the module so they can be overriden. Maybe they can be passed in via the `meta` map.

Reply:
Don't know how @maxulysse is passing read group info in sarek but worth seeing what he is doing there.

Reply:
Okay, working on it.